src/dotnet/APIView/APIViewWeb/Models/AssemblyModel.cs

```diff
@@ -0,0 +1,51 @@
+﻿using APIView;
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace APIViewWeb.Models
+{
+    public class AssemblyModel
+    {
+        public AssemblyModel(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        public int ID { get; set; }
+
+        [Required]
+        [Display(Name = "DLL Path")]
+        public string DllPath { get; set; }
+
+        [Display(Name = "Display String")]
+        public string DisplayString { get; set; }
+
+        // test DLL: C:\Users\t-mcpat\Documents\azure-sdk-tools\artifacts\bin\TestLibrary\Debug\netcoreapp2.1\TestLibrary.dll
+        public AssemblyModel()
+        {
+            this.DisplayString = "";
+            this.DllPath = "null";
+        }
+
+        public AssemblyModel(string dllPath, string fileName)
+        {
+            AssemblyAPIV assembly = null;
+            foreach (AssemblyAPIV a in AssemblyAPIV.AssembliesFromFile(dllPath))
+            {
+                if (fileName.EndsWith(".dll") && a.Name.Equals(fileName.Remove(fileName.IndexOf('.'))))
+                    assembly = a;
```

Comment:
Add a `break`?  Or tighten the whole thing up with a LINQ expression like 

`AssemblyAPIV assembly = AssemblyAPIV.AssembliesFromFile(dllPath).Where(a => fileName.EndsWith(".dll") && a.Name.Equals(fileName.Remove(fileName.IndexOf('.')))).FirstOrDefault()`

Reply:
Btw, the `System.IO.Path` class has a bunch of helpful methods like `Path.GetFileNameWithoutExtension(...)` and `Path.GetExtension(...)` as doing your own path manipulations can be kind of dangerous (i.e., you probably won't work with Azure.Storage.Blobs.dll since you're looking for the first dot).